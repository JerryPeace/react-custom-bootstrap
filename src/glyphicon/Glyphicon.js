import PropTypes from 'prop-types';
import React from 'react';
import bootstrapUtils from '../utils/bootstrapUtils';
import classNames from 'classnames';

class Glyphicon extends React.Component {

    static propTypes = {
        /**
         * icon type
         */
        bsStyle: PropTypes.string.isRequired,
        ariaHidden: PropTypes.bool
    };

    static defaultProps = {
        bsClass: 'glyphicon',
        ariaHidden: true
    };

    render() {
        const { className, ariaHidden } = this.props;
        const bsCls = bootstrapUtils.getCls(this.props);
        return (<span className={classNames(className, bsCls)} aria-hidden={ariaHidden} />);
    }
}


export default Glyphicon;
