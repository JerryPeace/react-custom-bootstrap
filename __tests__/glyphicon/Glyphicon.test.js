import React from 'react';
import { mount } from 'enzyme';
import Glyphicon from 'glyphicon/Glyphicon';

describe('Glyphicon', () => {
    test('render glyphicon correctly', () => {
        const glyphicon = mount(<Glyphicon/>);
        expect(glyphicon).toMatchSnapshot();
    });
});
