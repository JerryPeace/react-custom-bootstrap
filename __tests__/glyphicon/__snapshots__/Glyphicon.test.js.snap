// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glyphicon render glyphicon correctly 1`] = `
<Glyphicon
  ariaHidden={true}
  bsClass="glyphicon"
>
  <span
    aria-hidden={true}
    className="glyphicon"
  />
</Glyphicon>
`;
